File: backend/app/core/domain/wallet.py
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from decimal import Decimal
from enum import Enum
from uuid import UUID, uuid4


class WalletType(str, Enum):
    """Kind of money a wallet holds.

    CASH    — physical/held money
    DIGITAL — bank accounts, e-wallets, digital money
    SAVINGS — money set aside (family savings, marriage savings, ...)
    CREDIT  — credit cards / credit lines
    OTHER   — anything else
    """
    CASH = "cash"
    DIGITAL = "digital"
    SAVINGS = "savings"
    CREDIT = "credit"
    OTHER = "other"


@dataclass
class Wallet:
    user_id: UUID
    name: str
    type: WalletType = WalletType.CASH
    currency: str = "JOD"
    balance: Decimal = Decimal("0")
    initial_balance: Decimal = Decimal("0")
    icon: str | None = None
    is_active: bool = True
    id: UUID = field(default_factory=uuid4)
    created_at: datetime = field(default_factory=datetime.utcnow)

File: frontend/src/types/wallet.ts
export type WalletType = 'cash' | 'digital' | 'savings' | 'credit' | 'other'

export interface Wallet {
  id: string
  name: string
  type: WalletType
  currency: 'JOD' | 'USD'
  balance: string
  initial_balance: string
  icon: string | null
  is_active: boolean
  created_at: string
}

export interface WalletFormData {
  name: string
  type: WalletType
  currency: 'JOD' | 'USD'
  initial_balance?: string
  icon?: string
}

export interface TransferFormData {
  from_wallet_id: string
  to_wallet_id: string
  amount_original: string
  currency_original: 'JOD' | 'USD'
  transaction_date?: string
  description?: string
}

export const WALLET_TYPE_LABELS: Record<WalletType, string> = {
  cash: 'Held Money',
  digital: 'Digital Money',
  savings: 'Savings',
  credit: 'Credit',
  other: 'Other',
}

export const WALLET_TYPE_ICONS: Record<WalletType, string> = {
  cash: '💵',
  digital: '💳',
  savings: '🏦',
  credit: '🧾',
  other: '👛',
}
